wagtail/wagtailadmin/static_src/wagtailadmin/scss/components/_streamfield.scss

```diff
@@ -272,57 +298,83 @@ li.sequence-member {
         transform: translate3d(0, 0, 0) scale(1);
         visibility: visible;
         opacity: 1;
-        padding: 1em 1em 0;
+        padding: 1em 2em 1em;
+        display: flex;
+        flex-flow: row wrap;
         overflow: hidden;
     }
 
     li {
-        @include column(3);
-        padding-bottom: $grid-gutter-width;
-        padding-left: 0;
-        padding-right: 0;
+        box-sizing: border-box;
+        display: flex;
+        flex-direction: column;
+        float: none;
+        width: 8em;
+        padding: 1em 0;
+        text-align: left;
+        margin: 1em;
+    }
 
-        &:nth-child(4n+1) {
-            clear: left;
-        }
+    h3 {
+        padding: 0.5em 0;
+        margin: 2em 3em 0;
+        border-bottom: 1px solid $color-menu-text;
+        padding-bottom: 1em;
     }
 
     button {
-        @include transition(all 0.2s ease);
+        @include transition(all 0.4s ease);
         background-color: transparent;
         border: 0;
-        color: darken($color-grey-3, 5%);
+        color: $color-grey-1;
         height: auto;
         display: block;
-        width: 100%;
-        padding: 0 0 0.5em;
+        width: auto;
+        padding: 0;
         outline: $color-teal;
         overflow: visible;
 
         span {
-            text-transform: none;
+            text-transform: uppercase;
             white-space: pre-wrap;
             width: 100%;
-            display: block;
+            display: inline;
             overflow: hidden;
-            padding: 0 1em;
+            padding: 0.5em 0;
             box-sizing: border-box;
+            line-height: 0;
+            text-align: left;
+            margin: 0 5px 5px;
         }
 
-        &:before {
+        &::before {
             display: block;
             font-family: wagtail;
-            font-size: 2em;
-            width: 100%;
+            font-size: 1.2em;
+            width: 2em;
             height: 2em;
             line-height: 2em;
             text-align: center;
+            border: 1px solid $color-grey-3;
+            color: $color-grey-2;
+            border-radius: 8px;
+            margin: 0 auto 12px;
         }
```

Comment:
The block button is misaligned if there is no icon for it (see "Choices" on the right):

<img width="492" alt="streamfield-no-icon" src="https://user-images.githubusercontent.com/877585/33045409-5aa92c5e-ce55-11e7-9a91-75934ba229bd.png">

I'm not sure if this PR actually causes this, but could this be easily addressed? Otherwise if that's not a regression we should open a separate issue for this.

Reply:
Yup, it's a problem that - as far as I'm aware - has always been in Wagtail. It's certainly been in there since I started using wagtail 0.9. I think there _is_ an issue buried somewhere on it. If not will open one.